source/system/structs/VectorList.ooc

```diff
@@ -75,6 +75,55 @@ VectorList: class <T> extends List<T>{
 		(matches as Closure) free(Owner Receiver)
 		result
 	}
+	lowerBound: func (start, end: Int, toCompare: T, isLess: Func (T*, T*) -> Bool) -> Int {
+		version (safe)
+			if (!this isSorted(start, end, isLess))
+				raise("list must be partially sorted!")
+		count := end - start
+		current := start
+		while (count > 0) {
+			step := count / 2
+			current = start + step
+			compareResult: Bool
+			compareResult = isLess(this _vector[current]&, toCompare&)
+			if (compareResult) {
```

Comment:
Can't you merge the above three lines? Rock shouldn't have any trouble with that, right? Right? RIGHT? *so sad*